File: code_samples/scala/oop_encapsulation.scala
/**
 * Oop: Encapsulation
 * AI/ML Training Sample
 */

class Encapsulation {
  private var data: String = ""
  
  def process(input: String): Unit = {
    data = input
  }
  
  def getData: String = data
  
  def validate: Boolean = data.nonEmpty
}

object EncapsulationExample {
  def main(args: Array[String]): Unit = {
    val instance = new Encapsulation()
    instance.process("example")
    println(s"Data: ${instance.getData}")
    println(s"Valid: ${instance.validate}")
  }
}

File: code_samples/go/oop_encapsulation.go
package main

import (
    "fmt"
)

// Oop: Encapsulation
// AI/ML Training Sample

type Encapsulation struct {
    Data string
}

func NewEncapsulation() *Encapsulation {
    return &Encapsulation{
        Data: "",
    }
}

func (s *Encapsulation) Process(input string) {
    s.Data = input
}

func (s *Encapsulation) Validate() bool {
    return len(s.Data) > 0
}

func (s *Encapsulation) GetData() string {
    return s.Data
}

func main() {
    instance := NewEncapsulation()
    instance.Process("example")
    fmt.Printf("Data: %s\n", instance.GetData())
    fmt.Printf("Valid: %v\n", instance.Validate())
}
